src/components/expressions/expression-segment.tsx

```diff
@@ -1,27 +1,40 @@
 import React from 'react'
-import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native'
+import {
+	ColorValue,
+	StyleProp,
+	StyleSheet,
+	View,
+	ViewStyle,
+} from 'react-native'
 import { Text } from 'react-native-paper'
-import { LiteralValue, Method } from 'wollok-ts/dist/model'
+import { LiteralValue, Send } from 'wollok-ts/dist/model'
 import { OneOrMany } from '../../utils/type-helpers'
+import { getVisualSegment } from './ExpressionDisplay'
 
-export const ObjectSegment = (props: { text: string; index: number }) => {
+export const ReferenceSegment = (props: { text: string; index: number }) => {
 	return (
-		<Bullet color="green" index={props.index}>
+		<Pill index={props.index} color="#EF5B5B">
 			<Text>{props.text}</Text>
-		</Bullet>
+		</Pill>
 	)
 }
 
-export const MethodSegment = (props: { method: Method; index: number }) => {
+export const MessageSegment = (props: { send: Send; index: number }) => {
 	return (
-		<Bullet color="red" index={props.index}>
-			<View style={style.row}>
-				<Text>{props.method.name}</Text>
-				{props.method.parameters.map(({ name }) => (
-					<Parameter color="grey" key={name} text={name} />
-				))}
-			</View>
-		</Bullet>
+		<>
+			{getVisualSegment(props.send.receiver, props.index)}
+			<Bullet color="#4F518C" index={props.index + 1}>
+				<View style={style.row}>
+					<Text>{props.send.message}(</Text>
+					{props.send.args.map(a => (
+						<Parameter color="#907AD6">
+							{getVisualSegment(a, props.index - 1)}
+						</Parameter>
+					))}
```

Comment:
Acá no le falta una `key`?

Reply:
También se podría delegar la responsabilidad de saber cómo mostrar el segment al Parameter, sobre todo por saber qué hacer con el `index`:

```suggestion
					{props.send.args.map(a => (
						<Parameter color="#907AD6" arg={a} />
					))}
```